op-signer-mon/pkg/monitor/poll.go

```diff
@@ -174,38 +182,70 @@ func (p *Poller) pollRPC(ctx context.Context) (err error) {
 
 	jsonData, err := json.Marshal(reqBody)
 	if err != nil {
-		err = fmt.Errorf("error marshaling JSON", "err", err)
-		return
-	}
-	req, err := http.NewRequest("POST", endpoint, bytes.NewReader(jsonData))
-	if err != nil {
-		err = fmt.Errorf("error creating HTTP request", "err", err)
+		err = fmt.Errorf("error marshaling JSON: %w", err)
 		return
 	}
-	req.Header.Set("Content-Type", "application/json")
 
-	resp, err := client.Do(req)
-	if err != nil {
-		err = fmt.Errorf("HTTP request failed", "err", err)
-		return
-	}
-	defer resp.Body.Close()
+	var (
+		responseBody []byte
+		statusCode   int
+		lastErr      error
+	)
 
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		err = fmt.Errorf("rror reading response", "err", err)
-		return
+	backoff := rpcInitialBackoff
+
+	for attempt := 1; attempt <= rpcMaxAttempts; attempt++ {
+		select {
+		case <-ctx.Done():
+			err = ctx.Err()
+			return
+		default:
+		}
+
+		responseBody, statusCode, lastErr = performRPCRequest(ctx, client, endpoint, jsonData)
+		if lastErr == nil && statusCode == http.StatusOK {
+			break
+		}
+
+		shouldRetry := false
+		if lastErr != nil {
+			shouldRetry = isRetryableRPCError(lastErr)
+			lastErr = fmt.Errorf("HTTP request failed: %w", lastErr)
+			log.Debug(lastErr.Error())
+		} else {
+			lastErr = fmt.Errorf("unexpected status code: %d\nResponse: %s", statusCode, responseBody)
+			shouldRetry = isRetryableStatus(statusCode)
+			log.Debug(lastErr.Error())
+		}
+
+		if !shouldRetry || attempt == rpcMaxAttempts {
```

Comment:
```suggestion
		if !shouldRetry || attempt >= rpcMaxAttempts {
```
just being safe here?

Reply:
Ah. thats safe 👍